### ISSUE TITLE
Is this method simply doing segmentation of salient objects?

### ISSUE THREAD

**Issue by IssamLaradji** [MSiam/AdaptiveMaskedProxies#8]
At test time, I have ran this method with bird as support image and car as query image. The prediction was a segmentation mask over the car. 

This made me wonder, is the method simply doing segmentation of salient objects in the image? or am I doing something wrong?

Thanks.

**Comment by MSiam** [comment]
It is not (or at least not solely) for 2 simple reasons that we have validated in our experiments:

1- In the 1-way segmentation setup check Figure 5 in the paper, and you will find that it segmented the bike properly although it is definitely not the most salient object in the scene.

2- More importantly we actually have experiments for 2-way segmentation which proves without doubt its not doing salient object segmentation. Results are in the supplementary material not released yet. But you could see some qualitative results in the README.

Wouldn't you agree if it was simply doing salient object segmentation we would never be able to perform multi-class segmentation correct :) ? As for the test case you have tried I haven't done these experiments but its definitely interesting to check and understand more why this would happen.

**Comment by IssamLaradji** [comment]
that makes sense! thanks a lot for explaining :).

**Comment by MSiam** [comment]
Can you still send me the samples you had this issue with? Cause I am doing argmax on the logits while OSLSM was thresholding the probabilities with 0.5. So that simply can help solve that issue, I can look into that more after my deadline but please send me the samples you had this issue with.
Thanks
